core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/entity/descriptor/EntityDescriptor.java

```diff
@@ -602,6 +602,15 @@ public boolean hasNoNullVariables(Object entity) {
         return true;
     }
 
+    public boolean hasAnyNoNullGenuineAndShadowVariables(Object entity) {
```

Comment:
Consistently with `hasNoNullVariables()`, let's call this `hasNullVariables()`.

Reply:
It is not only about null variables or about genuine variables at all. It needs to take into account also empty list variables and null shadow variables as well. We don't won't to count them in. At first glance word "initialized" would sound good to me, but then I realized list variables are always considered initialized, so are the nullable entities.